pydantic/main.py

```diff
@@ -56,6 +56,15 @@
 _base_class_defined = False
 
 
+class _ModelNamespaceDict(dict):  # type: ignore[type-arg]
+    def __setitem__(self, k: str, v: object) -> None:
+        existing: Any = self.get(k, None)
+        if existing and v is not existing and isinstance(existing, _decorators.PydanticDecoratorMarker):
+            warnings.warn(f'`{k}` overrides an existing Pydantic `{existing.decorator_info.decorator_repr}` decorator')
```

Comment:
Do we care about the case of `elif existing and v is not existing and isinstance(v, _decorators.PydanticDecoratorMarker)`? I guess seems okay to not cover it.

Reply:
I don't think so. I don't want us to become a runtime linter lol, so I'm only covering the case of overriding a name with one of our decorators

Reply:
lol sounds good

Reply:
I guess the only thing I'll mention is that I think the original purpose of this was kind of runtime linting lol. So it might be a good idea to get Samuel's eyes on this and confirm he's in agreement. But I think we can merge with this decision and just change later.

Reply:
I mean it still does runtime linting, just in a limited fashion. cc @samuelcolvin